contracts/src/v0.8/dev/ArbitrumCrossDomainGovernor.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../interfaces/TypeAndVersionInterface.sol";
+import "./interfaces/DelegateForwarderInterface.sol";
+import "./vendor/arb-bridge-eth/v0.8.0-custom/contracts/libraries/AddressAliasHelper.sol";
+import "./vendor/openzeppelin-solidity/v4.3.1/contracts/utils/Address.sol";
+import "./CrossDomainDelegateForwarder.sol";
+
+/**
+ * @title ArbitrumCrossDomainGovernor - L1 xDomain account representation (with delegatecall support) for Arbitrum
+ * @notice L2 Contract which receives messages from a specific L1 address and transparently forwards them to the destination.
+ * @dev Any other L2 contract which uses this contract's address as a privileged position,
+ *   can be considered to be owned by the `l1Owner`
+ */
+contract ArbitrumCrossDomainGovernor is CrossDomainDelegateForwarder, TypeAndVersionInterface {
+  /**
+   * @notice creates a new Arbitrum xDomain Forwarder contract
+   * @param l1OwnerAddr tetherhe L1 owner address that will be allowed to call the forward fn
+   */
+  constructor(address l1OwnerAddr) CrossDomainDelegateForwarder(l1OwnerAddr) {
+    // noop
+  }
+
+  /**
+   * @notice versions:
+   *
+   * - ArbitrumCrossDomainGovernor 0.1.0: initial release
+   *
+   * @inheritdoc TypeAndVersionInterface
+   */
+  function typeAndVersion() external pure virtual override returns (string memory) {
+    return "ArbitrumCrossDomainGovernor 0.1.0";
+  }
+
+  /**
+   * @notice The L2 xDomain `msg.sender`, generated from L1 sender address
+   * @inheritdoc CrossDomainForwarder
+   */
+  function crossDomainMessenger() public view virtual override returns (address) {
+    return AddressAliasHelper.applyL1ToL2Alias(l1Owner());
+  }
+
+  /**
+   * @notice transfer ownership of this account to a new L1 owner
+   * @dev Forwarding can be disabled by setting the L1 owner as `address(0)`. Accessible only by the L1 owner (not the L2 owner.)
+   * @param to new L1 owner that will be allowed to call the forward fn
+   */
+  function transferL1Ownership(address to) external override {
```

Comment:
@krebernisak Was this pattern used to mitigate the risk of the bridge API breaking? (Since in the 2-step ownership transfer, the L1 owner has to accept ownership with a cross-chain message.)

Reply:
One side effect of this change is that this existing behaviour:

```
   * @dev Forwarding can be disabled by setting the L1 owner as `address(0)`.
```

has been removed, since the `acceptOwnership()` cannot be called from the zero address.